src/objprint/objprint.py

```diff
@@ -130,12 +132,16 @@ def objstr(self, obj: Any, **kwargs) -> str:
         return self._objstr(obj, memo, indent_level=0, cfg=cfg)
 
     def _objstr(self, obj: Any, memo: Optional[Set[int]], indent_level: int, cfg: _PrintConfig) -> str:
+        # If a custom formatter is registered for the object's type, use it directly
+        if cfg.inherit and self.type_formatter:
```

Comment:
To have the ``inherit`` option in the formatter itself, you need a data structure like ``{int: {"formatter": formatter, "inherit": True}}``. Check whether the exact type has a formatter first, then go through the mro and see if a base type has the formatter and inherit is True.

Reply:
I would prefer to use a namedtuple to store formatter and inherit, instead of a nested dictionary.